aptos/cli/src/commands/aptos.ts

```diff
@@ -0,0 +1,499 @@
+import { spawnSync } from "child_process";
+import { BCS, AptosClient, AptosAccount, TxnBuilderTypes } from "aptos";
+import type { CommandBuilder } from "yargs";
+import fs from "fs";
+import sha3 from "js-sha3";
+import { ethers } from "ethers";
+
+const LOCALNET: string = "localnet";
+const TESTNET: string = "testnet";
+const MAINNET: string = "mainnet";
+
+interface Network {
+  // RPC endpoint of the network
+  endpoint: string;
+  // Private key of the network
+  key: string | undefined;
+  // Address of the Pyth deployer contract
+  deployer: string;
+  // The Pyth deployer contract seed used to generate the derived address of the Pyth contract
+  pythDeployerSeed: string;
+}
+
+const network = {
+  alias: "n",
+  describe: "network",
+  type: "string",
+  choices: [LOCALNET, TESTNET, MAINNET],
+  required: true,
+} as const;
+
+interface Package {
+  meta_file: string;
+  mv_files: string[];
+}
+
+interface PackageBCS {
+  meta: Uint8Array;
+  bytecodes: Uint8Array;
+  codeHash: Uint8Array;
+}
+
+const networks = new Map<string, Network>([
+  [
+    LOCALNET,
+    {
+      key: process.env["APTOS_LOCALNET_KEY"],
+      endpoint: "http://0.0.0.0:8080",
+      deployer:
+        "0x277fa055b6a73c42c0662d5236c65c864ccbf2d4abd21f174a30c8b786eab84b",
+      pythDeployerSeed: "pyth",
+    },
+  ],
+  [
+    TESTNET,
+    {
+      key: process.env["APTOS_TESTNET_KEY"],
+      endpoint: "https://fullnode.testnet.aptoslabs.com/v1",
+      deployer:
+        "0xb31e712b26fd295357355f6845e77c888298636609e93bc9b05f0f604049f434",
+      // A Wormhole redeploy meant we had to use different seeds for testnet and mainnet
+      pythDeployerSeed: "pyth-a8d0d",
+    },
+  ],
+  [
+    MAINNET,
+    {
+      key: process.env["APTOS_MAINNET_KEY"],
+      endpoint: "https://fullnode.mainnet.aptoslabs.com/v1",
+      deployer:
+        "0xb31e712b26fd295357355f6845e77c888298636609e93bc9b05f0f604049f434",
+      pythDeployerSeed: "pyth-a8d0d",
+    },
+  ],
+]);
+
+export const command: string = "aptos <command>";
+
+export const builder: CommandBuilder = (yargs) =>
+  yargs
+    .command(
+      "deploy <package-dir> <account>",
+      "Deploy a package",
+      (yargs) => {
+        return yargs
+          .positional("package-dir", { type: "string" })
+          .positional("account", { type: "string" })
+          .option("named-addresses", { type: "string" })
+          .option("network", network);
+      },
+      async (argv) => {
+        const artefact = serializePackage(
```

Comment:
is there a specific reason it is `artefact`? :D

Reply:
I took inspiration from the phrase "build artefact", but better name suggestions are welcome 😄